### DISCUSSION TITLE
Populate initial browser window with random albums

### DISCUSSION THREAD

**Discussion by ghost** [SoongNoonien/mpdevil#91] [Ideas]
Hi
I know you have considered, and rejected, the concept of a random album feature in the past but I wondered if you might consider a less intrusive proposal.

Rather than pollute the UI with further buttons or configuration options I was thinking it might make good use of the initial browser window to be populated with, say 10, random albums rather than the user just seeing whatever is the first Artist in their collection as is currently the case. If a user is not interested in any of the random albums they would simply search for something they want as is the case currently. I'm not proposing a refresh button or any other UI changes.

Whatever your decision thank you for providing this project to the community.

**Comment by SoongNoonien** [comment]
Hi!
Sorry for the late reply. I'm still not a fan of randomness. It is very hard to implement some randomness which feels random and is still reasonably fast. To choose a truly random album I'd have to fetch all albums from mpd first and mpd's definition of an album slightly differs from mine so this would be relatively slow. Showing the first artist when nothing is playing on startup is not ideal but I don't think displaying random albums is a good solution to that.
One possible solution I've thought about would be to display some kind of colorful [placeholder page](https://developer.gnome.org/hig/patterns/feedback/placeholders.html) instead of the albums of the first artist, the artist list on the left would still be visible though. This page could say something like "Select an artist". What do you think about that?

**Reply by ghost** [reply]
I wouldn't want to slow mpdevil down! one of the attractions of mpdevil is how quick it is.

I agree a random album page is not in itself a good solution to viewing the first artist each time but I think a placeholder page may look attractive when first introduced but quickly become as repetitive as viewing the first artist.
My interest in a random album feature, however implemented, is due to having over 15k albums in my collection where albums and artist get forgotten about. A random album offers a possible starting point for that sessions listening or if I already know what I want I can just jump to that. I accept I may be in the minority here.

Slightly off topic but are there plans to re-introduce a direct artists column search in the GTK4 port ? I have 3K artists and scrolling a list that long is far too slow. I know I can get to the artist via a track search but that's quite a clunky way of getting there.

As always thank you for your continued work on this project and making it available to the community.

**Reply by SoongNoonien** [reply]
> I agree a random album page is not in itself a good solution to viewing the first artist each time but I think a placeholder page may look attractive when first introduced but quickly become as repetitive as viewing the first artist.

Yes, this could be the case.

> Slightly off topic but are there plans to re-introduce a direct artists column search in the GTK4 port ?

Yes, is some sense. Take a look at: #86

**Reply by rien333** [reply]
> One possible solution I've thought about would be to display some kind of colorful [placeholder page](https://developer.gnome.org/hig/patterns/feedback/placeholders.html) instead of the albums of the first artist, the artist list on the left would still be visible though. This page could say something like "Select an artist". What do you think about that?

Not a fan. I like the the fairly recent behavior, where mpdevil just shows me the albums by the artist I've just been playing. The behavior you are outlining would make sense in the context of something like "Gnome Music", where _starting_ the application means starting a new music listening session, but less so in a GUI frontend for a music player daemon, where starting the application may also mean modifying — rather than starting afresh — your session.

**Reply by SoongNoonien** [reply]
Yeah, good point.

**Reply by rien333** [reply]
> I agree a random album page is not in itself a good solution to viewing the first artist each time but I think a placeholder page may look attractive when first introduced but quickly become as repetitive as viewing the first artist.
My interest in a random album feature, however implemented, is due to having over 15k albums in my collection where albums and artist get forgotten about.

I think implementing different ways to sort the user's library is the way to go here. This, by the way, is also the way many other music players allow users to get different views of their library.  I was actually thinking of making UI mockups/contributing code to add  "sort by date added" behavior, which mpd provides. Adwaita also already has stuff for this, looking at the "EarTag" application: 

![Screenshot from 2024-03-02 13-52-02](https://github.com/SoongNoonien/mpdevil/assets/10417027/2720b3b8-026f-45ea-82e6-e0c028449ccd)

Sort by "Release year" seems like another good sorting option, and coupled with the ability to sort ascending/descending, that should give you some different ways to explore your library. (also @Richjn  I think mpd can track how many times you've played a song, maybe you could use that to find artists you still have to explore)

It's also good to remember that mpd itself is quite extensible.  For instance, using wofi, I've made some scripts that allow me to browse recently added albums and browse albums by genre, the latter of which interoperates a lot of randomization: 

[cast.webm](https://github.com/SoongNoonien/mpdevil/assets/10417027/6b72cd75-88da-4238-b086-d1c53b3fb42e)

**Reply by rien333** [reply]
~~Whoops, I might be wrong about mpd incorporating "date added" metadata, though I could easily live with using it's "date modified" field for this purpose.~~

@SoongNoonien [turns out](https://github.com/MusicPlayerDaemon/MPD/issues/172) mpd is going to support "date added" timestamps in the next release after all

**Reply by SoongNoonien** [reply]
I don't see how this could be implemented without maintaining at least a temporary copy of the whole database. Both metadata fields `last-modified` and `added` are metadata fields of individual songs and can't be used like normal tags is a `list` query (I mean the MPD command). And if that were possible artists that weren't added as a whole would appear on multiple positions in such a query. In the worst case one appearance per song.
But maybe I'm not fully getting your idea.

**Reply by ghost** [reply]
@rien333 Yes, there are several options available to get a different view of your mpd database.

I currently use ashuffle [https://github.com/joshkunz/ashuffle](url) to generate random albums / tracks for example. I understand I may be in the minority wanting a random album landing page and you did make a good point re: resuming a listening session.

I'm just hoping the artists list view issue arising from the changes in GTK4 can be resolved because mpdevil is painful to the point of not being useable, for me, currently with my 3k+ artists list.

**Comment by ghost** [comment]
Closing as feature is only supported by a minority of users.
